pytorch_lightning/utilities/parsing.py

```diff
@@ -247,9 +248,15 @@ def save_hyperparameters(
             hp = {arg: init_args[arg] for arg in args if isinstance(arg, str)}
             obj._hparams_name = "kwargs"
 
+    hp = obj._to_hparams_dict(hp)
+    # pop all nn.Module
+    for key in list(hp):
+        if isinstance(hp[key], nn.Module):
+            del hp[key]
```

Comment:
https://github.com/PyTorchLightning/pytorch-lightning/blob/52013f066641f07c8abb4a27dc0583447101afb9/pytorch_lightning/utilities/parsing.py#L234

Can we just add a check if key is not a `nn.Module` instance in the line above and avoid deleting?

Reply:
yep!